examples_skyline/helper.py

```diff
@@ -0,0 +1,53 @@
+import os 
+import sys
+import shutil
+import pandas as pd
+import numpy as np
+
+path = 'logs'
+def extract_info(learner_acc, learner_di, learner_fnr, learner_fpr, learner_label, learner_optimal):
+    try:
+        for folder_name, sub_folders, file_names in os.walk(path):
+            for sub_folder in sub_folders:
+                file_list = os.listdir(os.path.join(path, sub_folder))
+                file_list.sort()
+                for file in file_list:
+                    file_path = os.path.join(path,sub_folder, file)
+                    df = pd.read_csv(str(file_path), header=None, names = ['split_type', 'label', 'parameter', 'value'])
+                    test_data = df['split_type'] == 'test'
+                    
+                    flags = file.split("__")
+                    pre_processor = flags[1]
+                    post_processor = flags[2]
+ 
+                    if(len(flags) == 8 and flags[-1] == "OPTIMAL.csv"):
```

Comment:
please refactor this code into its own method, its really hard to understand what this is doing, so give the method a speaking name etc